### PR TITLE
Add support for Clarius OEM API

### PR DESCRIPTION
**Pull request by markasselin**
This PR:

- Adds support for the [Clarius OEM API](https://github.com/clariusdev/oem)
- Allows the user to select the CXX standard used to compile PLUS (test builds succeed with C++ 17, which is required for the Clarius OEM device which uses C++/WinRT)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,8 +5,9 @@ PROJECT(PlusBuild)
 # Configure the build to work (although with limited functionalities) if only
 # src directory of the repository is available
 
-SET(CMAKE_CXX_STANDARD 11 CACHE INTERNAL "Use C++11 standard")
-MARK_AS_ADVANCED(CMAKE_CXX_STANDARD)
+IF(NOT DEFINED CMAKE_CXX_STANDARD)
+  SET(CMAKE_CXX_STANDARD 11 CACHE STRING "C++ standard version")
```
Comment (adamrankin) [inline CMakeLists.txt]: I know we didn't before, but should this check also set `CMAKE_CXX_STANDARD_REQUIRED` ?

File: CMakeLists.txt [hunk 2]
```diff
@@ -448,6 +450,10 @@ IF(PLUS_USE_OPTITRACK)
   LIST(APPEND PLUSBUILD_QT_COMPONENTS Multimedia)
 ENDIF()
 
+IF(PLUS_USE_CLARIUS_OEM)
+  LIST(APPEND PLUSBUILD_QT_COMPONENTS Bluetooth)
```
Comment (adamrankin) [inline CMakeLists.txt:458]: Optional, but maybe a WIN32 check and warning if it's not a Windows build?

File: CMakeLists.txt (hunk 1 above)
Comment (lassoan) [reply CMakeLists.txt]: You could do what is done here: https://github.com/Slicer/ChestImagingPlatform/commit/bc173fdba7db8ebac9119bcef10dffaac20f7f03

Comment (markasselin) [reply CMakeLists.txt]: Sure, I've updated the PR to use something similar to what @lassoan suggested to manage the CMAKE_CXX_STANDARD value

File: CMakeLists.txt (hunk 2 above)
Comment (markasselin) [reply CMakeLists.txt:458]: Good idea! I added a fatal error message to warn the user that the ClariusOEM device is not available on non-Win32 platforms.

**Comment by markasselin** [comment]
Please hold off on integrating, I'm working on implementing automatic download of the OEM library via CMake ExternalProject

**Comment by markasselin** [comment]
This should be ready to integrate now, thank you for your suggestions!

**Pull request merged**